### PR TITLE
Add Italian strings for Preprocessor window

### PR DESCRIPTION
**Pull request by bovirus**
Add Italian strings for Preprocessor window installer option

### CODE REVIEW

**Comment by martijnlaan** [comment]
Isn't #268 a better version of this one? Did you mean to do this one on the ispack repo?

**Comment by bovirus** [comment]
This is the changes with accented letters ok (directlye edited by web browser)
Please close pullup request 268.

**Comment by martijnlaan** [comment]
Ok I will close it but I don't really understand: 268 contains more translations than this one.

**Comment by martijnlaan** [comment]
About this one: It's still in UTF8, see https://imgur.com/a/5sYYv3U

**Comment by bovirus** [comment]
@martijnlaan 

I'm so sorry but I didn't understand.
I didn't use and external editor.
I use directly the web browser editing the file via github with the browser.
I ddin't sue any kind of external editor like Notepad++ or PSPAd editor.

I believe that the .iss file "should be" an UTF-8 without BOM.

When I propose the changes via pullup request the comparing of original/new version file (befote make pull request) showed fine all accented letters.
Then I believe that could be a problem of github management about accented letters.

How can do without problem?

Then I didn't understand how can edit it without having probelm with accenetd letetrs.

**Comment by ajtruckle** [comment]
I find with BitBucket that it changes the UTF encoded of data on the website too. Perhaps Martijn can explain the method he uses for editing the source code files on GitHub.

**Comment by martijnlaan** [comment]
See my earlier comment in #267:
> Did you use the GitHub's web editor to make the changes? That doesn't seem to work indeed: it will auto change things into UTF8. So you'll need to use Git itself and a proper editor like Notepad or Notepad++. See https://github.com/jrsoftware/issrc/blob/master/CONTRIBUTING.md for an explanation about how to setup Git.

I do realize this is a lot more complicated. If you can come up with a nice way where the extra strings are located in a seperate file and where setup.iss will automatically include those we could make it work more like regular translation updates where doing it without Git is also possible without adding lots of extra work for me.

**Comment by ronaldhoek** [comment]
@bovirus : the UTF8 issue has nothing to do with the editor you're using locally on your machine, but with the fact you're committing the change to github using the GitHub website.

When you do this ALL TEXT you paste in the edit window, will be converted to Unicode (nog matter if you were using an ASCII/ANSI editor from which you copied the data).

See the warning when editing the original language file in the browser!

image

To create a correct pull request:
- make a fork of the project
- use a Git-client to clone the repository to your local machine
- edit the file on your local machine using an editor, whcih does nog convert it to unicode
- commit the changes to your fork en make a PULL request

**Comment by ronaldhoek** [comment]
@martijnlaan I was just to late finishing my reply ;)

**Comment by bovirus** [comment]
@ronaldhoek 

I'm sorry but is not my case.
Please check the screenshot.
Editing directly or edit the file by a fork of original project I didn't see any message about text code conversion.

issrc02
issrc01

The rpeview of changes show the accenetd letter right.

If I pass the strings to add could you for me?

The strings are

italian.ISPPTitle=Inno Setup Preprocessor
italian.ISPPSubtitle=Vuoi installare Inno Setup Preprocessor?
italian.ISPPText=Inno Setup Preprocessor (ISPP) è un addon ufficiale per Inno Setup. ISPP ti permette di compilare condizionalmente parti degli script, di usare variabili tempo negli script e di usare le funzioni integrate che per esempio possono leggere dal Registro o da file INI durante la compilazione.%n%nISPP contiene inoltre una speciale versione del compilatroe a linea di comando ISCC che può leggere i valori delle variabili come parametri della linea di comando e usarli durante la compilazione.
italian.ISPPText2=Seleziona se vuoi installare ISPP, quindi fai clic su 'Avanti'.
italian.ISPPCheck=&Installa Inno Setup Preprocessor

Thanks.

**Comment by ronaldhoek** [comment]
@bovirus When you've edited the code once and committed it on the GitHub website, you won't get the warning the second time, as it is already converted to UTF-8.

Regarding committing changes: I could change it, but then I would:
- OR need to have access write-access to your fork on GitHub
- OR need to create my own fork, and then create a PULL to your fork

**Comment by bovirus** [comment]
@ronaldhoek 

I didn't have write access to the repository but i can create directly a pullup request editing the code.

Otherwise if you prefer you ca make a fork and creating a pullup request on the fork.

Thanks.

**Comment by ronaldhoek** [comment]
@bovirus  and @martijnlaan 

Ok, but these changes are now in de 'setup.iss' file and not inside the 'Files\Languages\Italian.isl' file.
Also, I don't see this information for any other language in de 'Inno Setup' setup script or the corresponding '.isl' file.

As far as I can see, the setup of 'Inno Setup (pack)' itself is not language specific?

NOTE: the site of Jordan is down, so I could not check the regular installer (I only use the 'Pack' version)

**Comment by bovirus** [comment]
@ronaldhoek 

Thanks.
The string of Preprocessor are not in master English isl language then I didn't add to the Italian.isl file.

**Comment by bovirus** [comment]
@martijnlaan 

I agree with @ronaldhoek 
I believe that could be better to add to the master English .isl language 

strings for Inno Setup Preprocessor
strings for Quick Start Pack installer

then modify .iss strings (Inno Setup & Quick Start Pack .iss installer scripts) to point to CustomMessages of .isl language file.
If the translation is available in a specific language you can use it.
Otherwise will be used English strings.

**Comment by bovirus** [comment]
@martijnlaan 

I agree with you.

Could you modify the main .iss to point for extra string to a new file (ex. AdditionalStrings.isl)?
Could be the extra file a unique file that contains more than one language?
Or should be created several AdditionalStrings-xx.isl file (where xx is language abbreviation like en, it, etc)

We can do it also for Quick Start pack?
I believe to be nice to have not only Inno Script but also Quick Start pack localized.

Thanks.

**Comment by martijnlaan** [comment]
Yes, see a00c2a8c25050e2c3b82f52afea46b7531d02aab for what I was thinking about.

**Comment by bovirus** [comment]
@martijnlaan 

Thanks for the implementation. I believe that it's wonderful.

Than anyone that want to add the translation about Inno Setup Preprocessor window (and any future additional strings) can do that easily.
Without touching original master Inno Setup Language file.

Could you please apply the same things to Inno Setup Quick Start Pack?

Thanks.

**Comment by martijnlaan** [comment]
Please feel free to help with this. First step would be a pull request which changes all the strings that need translations in QSP's setup.iss into [CustomMessages] in setup.iss.

**Comment by bovirus** [comment]
If I make a pull request with all strings to change the you should re-adapt to the new structure of the program.

If make the changes to master QSO iss file to point all QSP additional master English strings to an external setup .isl file, and create QSP Setup English .isl file, I (and other translator that want to translate it) can post the localized version of new QSP addition setup isl file.

I think that could be easier for you and for us.

**Comment by martijnlaan** [comment]
Yes, I just meant to do it step by step. First stepping is adding a [CustomMessages] section to QSP's setup.iss.

**Comment by bovirus** [comment]
@martijnlaan 

Thanks again.
I will watch the QSP github section for step by step translation of custom messages.
About pre-processor window translation I believe that you can call the ticket.

File: setup.iss [hunk 1]
```diff
@@ -187,6 +187,11 @@ ISPPSubtitle=Would you like to install Inno Setup Preprocessor?
 ISPPText=Inno Setup Preprocessor (ISPP) is an official add-on for Inno Setup. ISPP allows you to conditionally compile parts of scripts, to use compile time variables in your scripts and to use built-in functions which for example can read from the registry or INI files at compile time.%n%nISPP also contains a special version of the ISCC command line compiler which can take variable definitions as command line parameters and use them during compilation.
 ISPPText2=Select whether you would like to install ISPP, then click Next.
 ISPPCheck=&Install Inno Setup Preprocessor
+italian.ISPPTitle=Inno Setup Preprocessor
+italian.ISPPSubtitle=Vuoi installare Inno Setup Preprocessor?
+italian.ISPPText=Inno Setup Preprocessor (ISPP) è un addon ufficiale per Inno Setup.%n%nISPP ti permette di compilare condizionalmnte parte degli script, usare variabili tempo compilazione negli script e usare le funzioni integrate che per esempio possono leggere il Registro o i file INI durante la compilazione.%n%nISPP include anche una versione speciale del compilatore a linea di comando ISCC che può acquisre il valore delle variabili e usarle durante la compilazione.
```
[the remark this answers is not in the document]
Comment (bovirus) [reply setup.iss]: I' afraid abut it's more or less the same phrases.
Please highlight with bold the difference with my proposal and why you prefer that.

**Comment by bovirus** [comment]
@martijnlaan 

Many thanks for the changes.
Could you please rebuild the Inno Setup installer to test the new installer with localized text fro Pre-processor window?

Could you please the same things to Quick Start Pack repository?

- Enable read of all languages available (now is only English)
- Crate a separate setup file for installer strings (to enable translator to translate all strings)

Thanks.

**Comment by martijnlaan** [comment]
> Could you please rebuild the Inno Setup installer to test the new installer with localized text fro Pre-processor window?

It works fine so making a build isn't really needed.

> Could you please the same things to Quick Start Pack repository?

I was hoping for your help for this. I don't know yet if / when I have time to do this myself. Like I said, first step is adding a [CustomMessages] section to QSP's setup.iss so feel free to do that.

**Pull request closed without merging**